include/xtypes/DynamicData.hpp

```diff
@@ -439,6 +650,214 @@ class WritableDynamicDataRef : public ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate WritableDynamicDataRef
+    class Iterator : public ReadableDynamicDataRef::Iterator
+    {
+    public:
+        Iterator(
+                WritableDynamicDataRef& ref)
+            : ReadableDynamicDataRef::Iterator(ref, false)
+        {}
+
+        WritableDynamicDataRef operator * ()
+        {
+            return ReadableDynamicDataRef::Iterator::operator*();
+        }
+
+    protected:
+
+        friend class WritableDynamicDataRef;
+
+        Iterator(
+                WritableDynamicDataRef& ref,
+                bool end)
+            : ReadableDynamicDataRef::Iterator(ref, end)
+        {
+        }
+    };
+
+    /// \brief Returns the initial iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The initial iterator.
+    Iterator begin()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this);
+    }
+
+    /// \brief Returns the final iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The final iterator.
+    Iterator end()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this, true);
+    }
+
+    class MemberPair : public ReadableDynamicDataRef::MemberPair
+    {
+    public:
+        MemberPair(
+                const Member& member,
+                uint8_t* data)
+            : ReadableDynamicDataRef::MemberPair(member, data)
+        {}
+
+        WritableDynamicDataRef data()
+        {
+            return WritableDynamicDataRef(member_.type(), data_);
+        }
+
+        WritableDynamicDataRef data() const
```

Comment:
I think that this breaks the constant consistency. A const access should not return a WritableReference. Change to Readable and call to the parent class functionality?